### DISCUSSION TITLE
RTK Query, Code Splitting, and Refresh on Focus

### DISCUSSION THREAD

**Discussion by ervwalter** [reduxjs/redux-toolkit#2353] [Q&A] [answered]
I am seeing that refreshOnFocus does not work in my application when my APIs are added via injectEndpoints as described [here](https://redux-toolkit.js.org/rtk-query/api/created-api/code-splitting).  

- My base API has no endpoints and all endpoints are added via injectEndpoints
- My redux store includes the middleware from the api (though it's added before the endpoints are injected)
- I am calling setupListeners after creating my store

Observed behavior is that when a tab gets focus, the API is not refetched.  I did an experiment where I moved my API endpoints into the createApi call instead of injecting it later and things started working.  

Is it expected behavior that refetchOnFocus doesn't work for injected endpoints?  Or am I doing something wrong maybe?  I didn't want to create an issue until I was sure this wasn't human error...

``` typescript
//
// api.ts
//
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

export const api = createApi({
	baseQuery: fetchBaseQuery({
		baseUrl: "/api/",
	}),
	endpoints: () => ({}),
	refetchOnMountOrArgChange: 60,
	refetchOnFocus: true,
	refetchOnReconnect: true
});

//
// store.ts
//
export const store = configureStore({
	reducer: {
		api: api.reducer,
	},
	middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware),
});

setupListeners(store.dispatch);

//
//games.ts
//
import { api } from "./api";
import { Game } from "../games";

const gamesApi = api.injectEndpoints({
	endpoints: (build) => ({
		games: build.query<Game[], void>({
			query: () => "games/topten",
		}),
	}),

});

export const { useGamesQuery } = gamesApi;
```

**Comment by phryneas** [comment, marked as answer]
That should totally be working - and from looking at the sources at https://github.com/reduxjs/redux-toolkit/blob/67d41a6f8c29603023f586a24689817cd66d2917/packages/toolkit/src/query/core/buildMiddleware/windowEventHandling.ts#L29 I don't really see a reason why it would not.

Could you provide a CodeSandbox or a Replay?

**Reply by ervwalter** [reply]
Sure, I'll replicate my setup in a CodeSandbox later this afternoon and post here.

**Reply by ervwalter** [reply]
Fun times.  I recreated my code in CodeSandbox (my project was pretty much still an empty template anyway, so it was easy), and I couldn't reproduce the problem.  So I went back to my real code to see what was different... and I can't reproduce it there either. 

Maybe this morning's behavior was a hallucination.  Certainly seemed real to me, though.

In any case, nevermind.  I'm going to chalk this up to unknown human error that got fixed when I was experimenting and refactoring code trying to figure out what was causing the problem.
